ui/app/src/services/content.ts

```diff
@@ -660,36 +660,40 @@ export function deleteItem(
   indexToDelete: number | string,
   path: string
 ): Observable<any> {
-  return performMutation(
-    site,
-    path,
-    (element) => {
-      let index = indexToDelete;
-      let fieldNode = element.querySelector(`:scope > ${fieldId}`);
-
-      if (typeof indexToDelete === 'string') {
-        index = parseInt(popPiece(indexToDelete));
-        // A fieldId can be in the form of `a.b`, which translates to `a > item > b` on the XML.
-        // In terms of index, since all it should ever arrive here is collection items,
-        // this assumes the index path points to the item itself, not the collection.
-        // By calling removeLastPiece(indexToDelete), we should get the collection node here.
-        fieldNode = extractNode(element, fieldId, removeLastPiece(`${indexToDelete}`));
-      }
+  return lock(site, path).pipe(
```

Comment:
A lock is needed before writing, the issue was not happening because there admin can unlock and it doesn't matter the lock owner. For author/publisher/etc, since it was not locking first, the lock owner wasn't itself then it was not being able to unlock.

Reply:
Where is the unlock call happening?

Reply:
performMutation does a `writeContent` with an `unlock` parameter set to true. So at the end of the writing process it unlocks the content

Reply:
Please look at previous releases and identify if this was a change in the API behaviour and report back.

Reply:
It seems like it hasn't changed. Is the same behavior since 4.0.0

Reply:
This probably breaks the zone menu. Did you test that?

Reply:
This might have been a good pattern if all worked this way. For now, we should stick to the way it works all over which is the consumer issues the lock apart, before issuing the op. Perhaps we should use the beforeWrite$.

Reply:
I wasn't breaking the ZoneMenu, though there were 2 lock calls. Updated for it to use beforeWrite$